EDP-380C-28-R2-Lab3.R

```diff
@@ -12,3 +12,500 @@ source("scripts/rmvnorm.R")
+set.seed(1237)
+n <- 100000
+
+x_p <- list(
+    mus = c(0,0,0,0,0),
+    sigmas = c(sqrt(1), sqrt(2), sqrt(3), sqrt(4), sqrt(5)),
+    rhos = rep(.15, 10)
+)
+
+y_p <- list(
+    mus = c(10),
+    sigmas = c(4),
+    rhos = c(-.15, -.5, .15, .30, .20)
+)
+
+method2Fun <- function(n, x_p, y_p, betas){
+    #get params into workable form
+    params <- transFun(x_p, y_p)
+    #gereate covrmtrix for x
+    covarMatrix <- genSigma(x_p)
+    # generate X
+    X <- rmvnorm(n, x_p$mus, covarMatrix)
+    #intercept
+    X <- cbind(matrix(1, NROW(X)), X)
+    # generate the covariance matrix, Sigma
+    covarMatrix <- genSigma(params)
+    # solve for the betas
+    betas <- solve(covarMatrix[-1, -1]) %*% covarMatrix[1, -1]
+    # determine the error variance 
+    varE <- covarMatrix[1,1] - t(betas) %*%  covarMatrix[1, -1]
+    # generate y
+    # expectation of Y
+    Ey <- params$mus[1] - t(params$mus[-1]) %*% betas
+    # generate y
+    Y <- Ey[1] + X[,-1] %*% betas + rnorm(n, 0, sqrt(varE))
+    
+    dat <- cbind(Y, X)
+    return(dat)
+}
+
+test2 <- method2Fun(100, x_p, y_p, betas = betas)
+method2Mod <- myLM(test2[,1], test2[,2:7])
+method2Mod
+
+#       Estimate        SE   t value      p value
+# b0    10.4172707 0.2514375 41.430851 3.514433e-62
+# b1    -0.9718173 0.2435858 -3.989630 1.309803e-04
+# b2    -1.4946464 0.1908604 -7.831097 7.208772e-12
+# b3     0.3657604 0.1443035  2.534661 1.290787e-02
+# b4     0.7347021 0.1324599  5.546599 2.667136e-07
+# b5     0.3593766 0.1170519  3.070234 2.796316e-03
+# SD(e)  2.4684309        NA        NA           NA
+# R2     0.5314705        NA        NA           NA
+
+# calculate R2
+# function to create the covariance matrix
+params <- transFun(x_p, y_p)
+# empty matrix
+corMatrix = matrix(NA, nrow = length(params$sigmas), ncol = length(params$sigmas))
+# put in the diagonal correlations which is always just 1s
+diag(corMatrix) <- rep(1, length(params$sigmas))
+# add in the correlations bw variables on the lower triangle
+corMatrix[lower.tri(corMatrix)] <- params$rhos
+#and the upper triangle
+corMatrix[upper.tri(corMatrix)] <- t(corMatrix)[upper.tri(t(corMatrix))]
+# get R2 to compare to estimate eq 27
+R2 <- corMatrix[-1, 1] %*% solve(corMatrix[-1, -1]) %*% corMatrix[1, -1]
+
+Ey <- params$mus[1] - t(params$mus[-1]) %*% betas
+# generate y
+Y <- Ey[1] + X[,-1] %*% betas + rnorm(n, 0, sqrt(varE))
+
+#gereate covrmtrix for x
+covarMatrix <- genSigma(params)
+# solve for the betas
+betas <- solve(covarMatrix[-1, -1]) %*% covarMatrix[1, -1]
+# determine the error variance 
+varE <- covarMatrix[1,1] - t(betas) %*%  covarMatrix[1, -1]
+
+pop <- c(Ey, betas, sqrt(varE), R2)
+estimate <- method2Mod[,1]
+difference <- pop - estimate
+
+difTab <- cbind(pop, estimate, difference)
+
+#               pop   estimate  difference
```

Comment:
The values below are off compared to the population parameters (which are correct).

Reply:
Check your sample size for the data.